first_ruby/Gemfile.lock

```diff
@@ -0,0 +1,225 @@
+GEM
```

Comment:
Whats this file for?

Reply:
Hold the gems dependency tree which include all versions for the app.

Reply:
what do you mean `gem dependency tree` ?
How is this different from `Gemfile`?

Reply:
`gem dependency tree` is the dependency from current Gemfile. It can be shown by using the command `gem dependency`.
`Gemfile.lock` is where the exact version of dependency is used. Where `bundler` will look up at `Gemfile.lock` to install
the exact same version if the project is loaded on another machine.
`Bundler` is used to provide consistent environment for Ruby project by tracking and installing the exact gems and versions needed.

Reply:
Bundler is an application to manage ruby Gemfile. There are other kinds but bundler is the most commonly used